### PR TITLE
Enable sending custom extension headers in Kyma eventing

### PR DESCRIPTION
**Pull request by montaro**
See also #3856

### CODE REVIEW

File: components/event-bus/internal/knative/util/kn-eventing.go [hunk 1]
```diff
@@ -301,10 +301,7 @@ func makeHttpRequest(channel *evapisv1alpha1.Channel, headers *map[string]string
 		return nil, err
 	}
 	req.Header.Set("Content-Type", "application/json")
-
-	for k, v := range *headers {
-		req.Header.Set(k, v)
-	}
+	req.Header = *headers
```
Comment (marcobebway) [inline components/event-bus/internal/knative/util/kn-eventing.go]: Please move this line before `req.Header.Set("Content-Type", "application/json")`, to not lose setting the content-type header.

Comment (montaro) [reply components/event-bus/internal/knative/util/kn-eventing.go]: Thanks for the catch!
Fixed

**Comment by sayanh** [comment]
@montaro can you please add a test which takes care of:
> In case the headers have the same name as the one used by kyma eventing e.g. CE-Event-type, the precedence should be given to the one derived from the payload.

**Comment by sayanh** [comment]
https://github.com/kyma-project/kyma/pull/4106#issuecomment-494412272
It's fine if we add it in a separate PR

**Pull request merged**